Managers/PerformanceMan.h

```diff
@@ -0,0 +1,215 @@
+
+		/// <summary>
+		/// Saves current absolute time in microseconds as a start of performance measurement.
+		/// </summary>
+		/// <param name="counter">Counter to start measurement for.</param>
+		void StartPerformanceMeasurement(PerformanceCounters counter);
+
+		/// <summary>
+		/// Saves current absolute time in microseconds as an end of performance measurement. The difference is added to the value of current performance sample.
+		/// </summary>
+		/// <param name="counter">Counter to stop and updated measurement for.</param>
+		void StopPerformanceMeasurement(PerformanceCounters counter);
+
+		/// <summary>
+		/// Sets the current ping value to display.
+		/// </summary>
+		/// <param name="ping">Ping value to display.</param>
+		void SetCurrentPing(unsigned int ping) { m_CurrentPing = ping; }
+#pragma endregion
+
+#pragma region Class Info
+		/// <summary>
+		/// Gets the class name of this Entity.
+		/// </summary>
+		/// <returns>A string with the friendly-formatted type name of this object.</returns>
+		virtual const std::string & GetClassName() const { return c_ClassName; }
+#pragma endregion
+
+	protected:
+
+		static const std::string c_ClassName; //!< A string with the friendly-formatted type name of this object.
+		static constexpr unsigned short c_MSPFAverageSampleSize = 10;
+		static constexpr unsigned short c_MaxSamples = 120; //!< How many performance samples to store, directly affects graph size.
+		static constexpr unsigned short c_Average = 10; //!< How many samples to use to calculate average value displayed on screen.
+
+		const unsigned short c_StatsOffsetX = 17; //!< Offset of the stat text from the left edge of the screen.
+		const unsigned short c_StatsHeight = 14; //!< Height of each stat text line.
+		const unsigned short c_GraphsOffsetX = 14; //!< Offset of the graph from the left edge of the screen.
+		const unsigned short c_GraphsStartOffsetY = 134; //!< Position the first graph block will be drawn from the top edge of the screen.
+		const unsigned short c_GraphHeight = 20; //!< Height of the performance graph.
+		const unsigned short c_GraphBlockHeight = 34; //!< Height of the whole graph block (text height and graph height combined).
+
+		bool m_ShowPerfStats; //!< Whether to show performance stats on screen or not.
+		bool m_AdvancedPerfStats; //!< Whether to show performance graphs on screen or not.
+
+		Timer *m_FrameTimer; //!< Timer for measuring millisecs per frame for performance stats readings.
+		float m_SimSpeed; //!< The simulation speed over real time.
+		unsigned short m_Sample; //!< Sample counter.
+
+		std::deque<unsigned int> m_MSPFs; //!< History log of readings, for averaging the results.
+		unsigned short m_MSPFAverage; //!< The average of the MSPF reading buffer above, computer each frame.
+		unsigned short m_CurrentPing; //!< Current ping value to display on screen.
+
+		std::string m_PerfCounterNames[PERF_COUNT]; //!< Performance counter's names displayed on screen.
```

Comment:
I really dislike these names. They're meaningless and neither they nor their comments really explain what they do or what they're for. If possible, try to fix this so they explain themselves better.

Reply:
Confuse.

Reply:
Me too lol.

I think I was talking about PerfData and PerfMeasureStart and Stop. I think I commented before reading through the cpp, they make more sense in context of what they do. I still don't love any of the naming in this file, but it's fine.